qemu/qemu.go

```diff
@@ -704,6 +704,77 @@ func (blkdev BlockDevice) QemuParams(config *Config) []string {
 	return qemuParams
 }
 
+// VhostUserDeviceType is a qemu networking device type.
+type VhostUserDeviceType string
+
+const (
+	//SCSI
+	VhostUserSCSI = "vhost-user-scsi-pci"
+	//net
+	VhostUserNet = "virtio-net-pci"
+)
+
+// VhostUserDevice represents a qemu vhost-user network device meant to be passed
+// in to the guest
+type VhostUserDevice struct {
+	SocketPath    string //path to vhostuser socket on host
+	CharDevID     string
+	TypeDevID     string //id (SCSI) or netdev (net) device parameter
+	MacAddress    string //only valid if device type is  VhostUserNet
+	VhostUserType VhostUserDeviceType
+}
+
+// Valid returns true if there is a valid socket path defined for VhostUserDevice
+func (vhostuserDev VhostUserDevice) Valid() bool {
+	if vhostuserDev.SocketPath == "" || vhostuserDev.CharDevID == "" ||
+		vhostuserDev.TypeDevID == "" ||
+		(vhostuserDev.VhostUserType == VhostUserNet && vhostuserDev.MacAddress == "") {
+		return false
+	}
+
+	return true
+}
+
+func (vhostuserDev VhostUserDevice) QemuParams(config *Config) []string {
+	var qemuParams []string
+	var charParams []string
+	var netParams []string
+	var devParams []string
+
+	charParams = append(charParams, "socket")
+	charParams = append(charParams, fmt.Sprintf(",id=%s", vhostuserDev.CharDevID))
```

Comment:
If you're going to use strings.Join to convert this slice into a string you might as well let it add the ','.  So below you'd have

strings.Join(charParams,",")

Reply:
This comment applies to netParams and devParams as well.

Reply:
@markdryan good call -- done